Branch-SDK/src/io/branch/referral/ServerRequestCreateUrl.java

```diff
@@ -119,6 +119,17 @@ public void onRequestSucceeded(ServerResponse resp, Branch branch) {
         }
     }
 
+    /**
+     * Calls the callback with the URL. This should be called on finding an existing url
+     * up on trying to create a URL asynchronously
+     * @param url
+     */
+    public void onUrlAvailable(String url){
```

Comment:
I don't understand the flow there. You're manually setting it on the request so that it can call the callback, but it's sync, so it's just getting returned anyway?